c10/core/TensorImpl.h

```diff
@@ -840,6 +840,10 @@ struct C10_API TensorImpl : public c10::intrusive_ptr_target {
     return key_set_.has(DispatchKey::XLA);
   }
 
+  bool is_lazy() const {
```

Comment:
this makes me wonder about how 'graduate' backends like XLA should behave in generic code.  If someone writes 'is_lazy()' shouldn't it return true for an XLA backend too? Or is that just a misinterpretation of what `is_lazy` means, and it is fine the way it is?

Reply:
I think this plays into whether or not we consider it a feature vs. an implementation detail that different plugins/vendors are implemented using the same LazyTensor machinery.

I can't really picture that interpretion of `is_lazy()` being useful as a frontend API, unless you're doing something with multiple lazy + eager backends, and want to generically bucket them or something.

It seems kinda reasonable to me to say that once a backend has graduated, like XLA, then in the frontend API we should refer to it specifically, and not by the generic flavor of backends that it's similar to.

Reply:
FWIW I don't like these APIs and replicated them mechanically and reluctantly. I'd rather not have them at all, if you agree. These helpers aren't useful.

Reply:
@asuhan then just delete it! No one is forcing you to add this method :P